src/database.ts

```diff
@@ -1962,18 +1971,27 @@ class Database extends GrpcServiceObject {
         : {};
 
     const getWriteSession = this.pool_.getWriteSession.bind(this.pool_);
-    const [session, transaction] = await promisify(getWriteSession)();
-    const runner = new AsyncTransactionRunner<T>(
-      session,
-      transaction,
-      runFn,
-      options
-    );
-
-    try {
-      return await runner.run();
-    } finally {
-      this.pool_.release(session);
+    // Loop to retry 'Session not found' errors.
+    while (true) {
```

Comment:
Should we have a sleep interval before retrying? Also, is it expected to retry it indefinitely?

Reply:
Correct, there's no need to wait, as there's no (known) contention on the backend. The client should just try to get a valid session and execute the transaction as quickly as possible. It should not be possible that it will retry indefinitely, as it will first try with all the sessions that are available in the pool, and once those have all been tried, it will create a new session.